### PR TITLE
Allow closing of drop on click

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Added close on click method and prop to drop component.

## Motivation
We can now use drop for select without the need to create its own component.

### CODE REVIEW

File: src/elements/OCDrop.vue [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+  <div :id="$_ocDrop_id" :uk-drop="$_ocDrop_props" @click="$_ocDrop_close">
     <div v-if="$slots.default" class="uk-card uk-card-default uk-card-small uk-card-body">
       <slot />
     </div>
     <slot v-else name="special" />
   </div>
 </template>
 <script>
+import UiKit from "uikit"
+import { uniqueId as _uniqueId } from "lodash"
```
Comment (DeepDiver1975) [inline src/elements/OCDrop.vue]: is it really necessary to import lodash?

File: src/elements/OCDrop.vue [hunk 2]
```diff
@@ -1,12 +1,15 @@
 <template>
-  <div :uk-drop="$_ocDrop_props">
+  <div :id="$_ocDrop_id" :uk-drop="$_ocDrop_props" @click="$_ocDrop_close">
```
Comment (DeepDiver1975) [inline src/elements/OCDrop.vue]: I'd really love to see this more in a vue.js manner to be implemented:
- drop is visible/hidden using v-if or v-show
- clicking the button changes the state upon the drop is shown

We need to get rid of this uikit magic .. one day ...

Comment (LukasHirt) [reply src/elements/OCDrop.vue]: Agree.

File: src/elements/OCDrop.vue (hunk 1 above)
Comment (LukasHirt) [reply src/elements/OCDrop.vue]: I haven't found an alternative for the same function outside of downloading some other packages. One way could be using random number but there is the chance of generating two times the same number and then 💥 There is this function exported outside of lodash and distributed as own package but I'm not sure if we want to get it this way.

Comment (DeepDiver1975) [reply src/elements/OCDrop.vue]: Bloody uikit

Comment (DeepDiver1975) [reply src/elements/OCDrop.vue]: What about setting the ID via a property and leave it to the Developer who uses this component?
Default could be just oc-drop-default-id or Something ....

Comment (LukasHirt) [reply src/elements/OCDrop.vue]: Sounds reasonable... Will do :wink:

Comment (LukasHirt) [reply src/elements/OCDrop.vue]: Done

**Pull request merged**